# UPEC vs ABU GWAS analysis

The goal of this project is to replicate bgWAS analysis from the below study.

***[Horizontally acquired papGII-containing
pathogenicity islands underlie the emergence of
invasive uropathogenic Escherichia coli lineages](https://www.nature.com/articles/s41467-020-19714-9)***


***Sequencing Data***

```/scratch/esnitkin_root/esnitkin/apirani/Project_Ecoli_ABU/Sequence_data/fastq/Public_data/UTI_induced_bacteremia/Downloads_2022_01_12```

***Assembly annotations***

```/nfs/esnitkin/Project_Ecoli_ABU/Sequence_data/assembly_annotation/```

### How the data was downloaded:

- Arwen provided a metadata sheet that she extracted from microreact website. The sheet contains various data but the relevant columns such as sample id, Phenotype1, Phenotype2 and Biosample is printed below.

`/gpfs/accounts/esnitkin_root/esnitkin/apirani/Project_Ecoli_ABU/Sequence_data/fastq/Public_data/UTI_induced_bacteremia/Downloads_2022_01_12/2022_microreact-project-O4QAYAJWw-data-fixed.csv`

- Using the Biosample acccessions the sequencing data using Biosample acessions provided by Arwen in Metadata sheet:


  


In [5]:
import os
import readline
import argparse
from itertools import islice
import subprocess
import numpy as np
import pandas as pd
pd.set_option('display.max_rows', None)
metadata_df = pd.read_csv("/gpfs/accounts/esnitkin_root/esnitkin/apirani/Project_Ecoli_ABU/Sequence_data/fastq/Public_data/UTI_induced_bacteremia/Downloads_2022_01_12/2022_microreact-project-O4QAYAJWw-data-fixed.csv", sep=',', header=0)
#print (metadata_df[['id', 'Phenotype1', 'Phenotype2', 'Biosample']]).head()
print (metadata_df.head)

<bound method NDFrame.head of                 id __latitude __longitude  year   Geographical origin  \
0          10-bcus      50.91        -1.4  2016       Southampton, UK   
1          11-bcus      50.91        -1.4  2016       Southampton, UK   
2          12-bcus      50.91        -1.4  2016       Southampton, UK   
3          13-bcus      50.91        -1.4  2016       Southampton, UK   
4          14-bcus      50.91        -1.4  2016       Southampton, UK   
5          15-bcus      50.91        -1.4  2016       Southampton, UK   
6          16-bcus      50.91        -1.4  2016       Southampton, UK   
7          17-bcus      50.91        -1.4  2016       Southampton, UK   
8          18-bcus      50.91        -1.4  2016       Southampton, UK   
9          19-bcus      50.91        -1.4  2016       Southampton, UK   
10          1-bcus      50.91        -1.4  2016       Southampton, UK   
11         21-bcus      50.91        -1.4  2016       Southampton, UK   
12         22-bcus   

***Number of UPEC and ABU samples with Phenotypes***

In [2]:
#### Number of UPEC and ABU samples
upec_abu_phenotype = ['invasive UPEC', 'ABU', 'cystitis']
metadata_df_Upec_ABU = metadata_df.loc[metadata_df['Phenotype1'].isin(upec_abu_phenotype)]
print ("\n\nNumber of UPEC and ABU samples (Non-feacal) - %s\n\n" % metadata_df_Upec_ABU.shape[0])
print (metadata_df_Upec_ABU[['id', 'Phenotype1', 'Phenotype2', 'Biosample']]).head()



Number of UPEC and ABU samples (Non-feacal) - 722


                id     Phenotype1                 Phenotype2     Biosample
0          10-bcus  invasive UPEC  urinary-source bacteremia    SAMC048175
1          11-bcus  invasive UPEC  urinary-source bacteremia    SAMC048176
2          12-bcus  invasive UPEC  urinary-source bacteremia    SAMC048177
3          13-bcus  invasive UPEC  urinary-source bacteremia    SAMC048178
4          14-bcus  invasive UPEC  urinary-source bacteremia    SAMC048179
5          15-bcus  invasive UPEC  urinary-source bacteremia    SAMC048180
6          16-bcus  invasive UPEC  urinary-source bacteremia    SAMC048181
7          17-bcus  invasive UPEC  urinary-source bacteremia    SAMC048182
8          18-bcus  invasive UPEC  urinary-source bacteremia    SAMC048183
9          19-bcus  invasive UPEC  urinary-source bacteremia    SAMC048184
10          1-bcus  invasive UPEC  urinary-source bacteremia    SAMC048165
11         21-bcus  invasive UPEC  urinary-sou

### Pan-genome analysis

Notes: 

- I ran Panaroo QC scripts on the assemblies and the results and my thoughts can be found in this notebook.

```/nfs/esnitkin/Project_Ecoli_ABU/Analysis/2021_08_12_Panaroo/2021_08_12_Panaroo/Panaroo Pre and Post processing.ipynb```

The QC plots and results are located here: 

```/nfs/esnitkin/Project_Ecoli_ABU/Analysis/2021_08_12_Panaroo_QC```

***Panaroo results for UPEC and ABU samples (n=722, excluding Faecal) samples are located here:***

```/nfs/esnitkin/Project_Ecoli_ABU/Analysis/2021_08_12_Panaroo/2021_08_12_Panaroo_exclude_fecal/```


In [11]:
pg_summary = pd.read_csv("/nfs/esnitkin/Project_Ecoli_ABU/Analysis/2021_08_12_Panaroo/2021_08_12_Panaroo_exclude_fecal/summary_statistics.txt", sep='\t', header=0)
print ("Pangenome Summary statistics\n\n")
print (pg_summary)

Pangenome Summary statistics


        Core genes (99% <= strains <= 100%)   2958
0  Soft core genes   (95% <= strains < 99%)    409
1      Shell genes   (15% <= strains < 95%)   2992
2      Cloud genes    (0% <= strains < 15%)  19735
3      Total genes  (0% <= strains <= 100%)  26094


***Roary results for UPEC and ABU samples (n=722, excluding Faecal) samples are located here:***
    
    
```/nfs/esnitkin/Project_Ecoli_ABU/Analysis/2021_09_17_roary```


In [12]:
roary_pg_summary = pd.read_csv("/nfs/esnitkin/Project_Ecoli_ABU/Analysis/2021_09_17_roary/summary_statistics.txt", sep='\t', header=0)
print ("Pangenome Summary statistics\n\n")
print (roary_pg_summary)

Pangenome Summary statistics


        Core genes (99% <= strains <= 100%)   2208
0  Soft core genes   (95% <= strains < 99%)    472
1      Shell genes   (15% <= strains < 95%)   3617
2      Cloud genes    (0% <= strains < 15%)  44721
3      Total genes  (0% <= strains <= 100%)  51018


Notes:

- Compared Panaroo and Roary results. 
- There were some differences in both the analysis - for example Panaroo found 30489 total genes compared to Roary’s 37,717 genes (Panaroo core: 2912; Roary core: 3067). 
- The paper used Roary + treeWAS and we intended to use Panaroo + Hogwash and check if we get the same hits.
- Last thing where I left off was - I was working on reproducing GWAS analysis performed in the paper (Roary + treewas ) (the paper excluded fecal samples from Pan-GWAS studies due to their unknown urinary phenotype).

### Variant calling analysis

- The scripts to run variant calling and results are located here: 

```/scratch/esnitkin_root/esnitkin/apirani/Project_Ecoli_ABU/Analysis/2022_01_17_UPEC_ABU_variant_calling/```

